torch/utils/data/dataset.py

```diff
@@ -185,6 +186,22 @@ def class_function(cls, source_dp, *args, **kwargs):
         function = functools.partial(class_function, cls_to_register)
         IterableDataset.functions[function_name] = function
 
+    def __reduce_ex__(self, *args, **kwargs):
+        print('called __reduce_ex', self)
+        if IterableDataset.reduce_ex_hook is not None:
+            print('called overridden __reduce_ex', self)
+            try:
+                return IterableDataset.reduce_ex_hook(self)
+            except NotImplementedError:
+                pass
+        return super().__reduce_ex__(*args, **kwargs)
```

Comment:
TODO: we need to add doc for user to make sure all connected (source) `DataPipe`s are saved within `__dict__`.

Reply:
I was thinking to have some sort of tests template, which will make sure that any pipe can be serialized, deserialized etc.